src/main/java/datawave/microservice/authorization/postgres/PostgresDatawaveUserLookup.java

```diff
@@ -0,0 +1,218 @@
+import datawave.security.authorization.DatawaveUser.UserType;
+import datawave.security.authorization.SubjectIssuerDNPair;
+
+/**
+ * A helper class to allow calls to be cached using Spring annotations. Normally, these could just be methods in {@link PostgresDatawaveUserService}. However,
+ * Spring caching works by wrapping the class in a proxy, and self-calls on the class will not go through the proxy. By having a separate class with these
+ * methods, we get a separate proxy that performs the proper cache operations on these methods.
+ */
+@CacheConfig(cacheNames = CACHE_NAME)
+public class PostgresDatawaveUserLookup {
+    public final Logger logger = LoggerFactory.getLogger(getClass());
+    private final PostgresDULProperties postgresDULProperties;
+    
+    public PostgresDatawaveUserLookup(PostgresDULProperties postgresDULProperties) {
+        this.postgresDULProperties = postgresDULProperties;
+    }
+    
+    @Cacheable(key = "#dn.toString()")
+    public DatawaveUser lookupUser(SubjectIssuerDNPair dn) {
+        return buildUser(dn);
+    }
+    
+    @Cacheable(key = "#dn.toString()")
+    public DatawaveUser reloadUser(SubjectIssuerDNPair dn) {
+        return buildUser(dn);
+    }
+    
+    private DatawaveUser buildUser(SubjectIssuerDNPair dn) {
+        logger.info("Inside postgres build user...");
+        logger.info(String.format("Database Address:%s", postgresDULProperties.getDbAddr()));
+        logger.info(String.format("Database Name:%s", postgresDULProperties.getDbName()));
+        UserType userType = UserType.USER;
+        if (postgresDULProperties.getServerDnRegex() != null && Pattern.matches(postgresDULProperties.getServerDnRegex(), getCN(dn.subjectDN()))) {
+            userType = UserType.SERVER;
+        }
+        
+        List<String> auths = new ArrayList<String>();
+        List<String> roles = new ArrayList<String>();
+        
+        try (Connection connection = getConnection()) {
+            logger.info("Got connection, performing query...");
+            
+            int personGuid = getPersonGuid(connection, dn, userType);
+            
+            // no user found matching the cert, cannot continue
+            if (personGuid == -1) {
+                return DatawaveUser.ANONYMOUS_USER;
+            }
+            
+            List<String> myAuths = getAuths(connection, personGuid);
+            auths.addAll(myAuths);
+            
+            // Table/list of Datawave roles associated with the user
+            if (postgresDULProperties.isRolesEnabled()) {
+                List<String> myRoles = getRoles(connection, personGuid);
+                roles.addAll(myRoles);
+            }
+            
+        } catch (SQLException e) {
+            logger.warn(e.toString());
+        }
+        
+        long creationTime = System.currentTimeMillis();
+        long expireTime = creationTime + 60000;
+        return new DatawaveUser(dn, userType, "", auths, roles, null, creationTime, expireTime);
```

Comment:
I think I asked this last time but what's the blank string for the third argument? Can we get a comment stating it so I don't ask it next time you show me this code.

Reply:
Good to go